csrc/device_lower/utils.h

```diff
@@ -375,6 +375,13 @@ Val* proveLinearAndGetStride(
     const ValGroup& linear_g,
     const ValGroups& domain);
 
+// Given `from`, project it to `to`. This function will return a subset of `to`
```

Comment:
nit: should this be in this util file? Why not `val_graph_visitor.h`?

Reply:
This function is removed